File: 백준/Silver/15650. N과 M （2）/N과 M （2）.py
def dfs(x):
    if len(result) == M:
        print(*result)
        return

    for i in range(x, N + 1):
        if i not in result:
            result.append(i)
            dfs(i)
            result.pop()

N, M = map(int,input().split())
result = []
dfs(1)

File: 백준/Silver/15650. N과 M （2）/N과 M （2）.cc
#include <iostream>
#include <vector>
using namespace std;

int N, M;
int visited[9] = {0, };
int result[9] = {0, }; 

void dfs(int x, int cnt)
{
    if (cnt == M) {
        for (int i=0; i<M; i++) {
            cout << result[i] << " ";
        }
        cout << "\n";
        return;
    }
    
    for (int i=x; i<=N; i++) {
        if (visited[i] == 0) {
            visited[i] = 1;
            result[cnt] = i;
            dfs(i+1, cnt+1);
            visited[i] = 0;
        }
    } 
}

int main()
{
    cin >> N >> M;
    dfs(1, 0);
}
